temporalio/worker/_nexus.py

```diff
@@ -0,0 +1,475 @@
+                    # TODO(nexus-prerelease): unify this failure completion with the path above
+                    err = TypeError(
+                        "Operation start method must return either nexusrpc.handler.StartOperationResultSync "
+                        "or nexusrpc.handler.StartOperationResultAsync"
+                    )
+                    handler_err = _exception_to_handler_error(err)
+                    return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                        task_token=task_token,
+                        error=temporalio.api.nexus.v1.HandlerError(
+                            error_type=handler_err.type.value,
+                            failure=await self._exception_to_failure_proto(
+                                handler_err.__cause__
+                            ),
+                            retry_behavior=(
+                                temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_RETRYABLE
+                                if handler_err.retryable
+                                else temporalio.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_NON_RETRYABLE
+                            ),
+                        ),
+                    )
+
+                return temporalio.bridge.proto.nexus.NexusTaskCompletion(
+                    task_token=task_token,
+                    completed=temporalio.api.nexus.v1.Response(start_operation=op_resp),
+                )
+
+        try:
+            completion = await run()
+            await self._bridge_worker().complete_nexus_task(completion)
+        except Exception:
+            temporalio.nexus.logger.exception("Failed completing Nexus operation")
+        finally:
+            try:
+                del self._running_operations[task_token]
+            except KeyError:
+                temporalio.nexus.logger.exception(
+                    "Failed to remove completed Nexus operation"
+                )
+
+    async def _handle_cancel_operation(
+        self, request: temporalio.api.nexus.v1.CancelOperationRequest, task_token: bytes
+    ) -> None:
+        temporalio.nexus.handler._current_context.set(
+            temporalio.nexus.handler._Context(
+                client=self._client,
+                task_queue=self._task_queue,
+                service=request.service,
+                operation=request.operation,
+            )
+        )
+        ctx = nexusrpc.handler.CancelOperationContext(
+            service=request.service,
+            operation=request.operation,
+        )
+        # TODO(nexus-prerelease): header
+        try:
+            await self._handler.cancel_operation(ctx, request.operation_token)
+        except Exception as err:
+            temporalio.nexus.logger.exception(
+                "Failed to execute Nexus operation cancel method", err
+            )
```

Comment:
We need to complete the nexus task with a failure if the call failed

Reply:
PTAL again, I believe the implementation satisfies requirements.